src/preamble.js

```diff
@@ -2212,7 +2224,9 @@ function integrateWasmJS(Module) {
 #if BINARYEN_ASYNC_COMPILATION
     Module['printErr']('asynchronously preparing wasm');
     addRunDependency('wasm-instantiate'); // we can't run yet
-    WebAssembly.instantiate(getBinary(), info).then(function(output) {
+    getBinaryPromise().then(function(binary) {
```

Comment:
Not as it currently works - the html downloads the wasm binary *before* starting to do anything else. Another reason it's good to get rid of that thing ;)